lib/managers/terms-of-service.js

```diff
@@ -0,0 +1,307 @@
+ * Method: GET
+ *
+ * @param {string} termsOfServicesID - The ID of the terms of services to retrieve status on
+ * @param {Object} [options] - Additional options. Can be left null in most cases
+ * @param {string} [options.userID] - Optional, the id of the user to retrieve status of custom terms and service on
+ * @param {Function} [callback] - Passed the terms of service user status information if successful, error otherwise
+ * @returns {Promise<Object>} A promise resolving to the terms of service user status
+ */
+TermsOfService.prototype.getUserStatuses = function(termsOfServicesID, options, callback) {
+	var params = {
+		qs: {
+			tos_id: termsOfServicesID
+		}
+	};
+
+	if (options) {
+		Object.assign(params.qs, options);
+	}
+
+	var apiPath = urlPath(USER_STATUSES_PATH);
+	return this.client.wrapWithDefaultHandler(this.client.get)(apiPath, params, callback);
+};
+
+/**
+ * Accepts/rejects custom terms of services for the user
+ *
+ * API Endpoint: '/terms_of_service_user_statuses'
+ * Method: PUT
+ *
+ * @param {string} termsOfServiceUserStatusesID - Terms of service user status object ID
+ * @param {bool} isAccepted - Determines wehether the terms of services has been accepted or rejected
+ * @param {Function} [callback] - Passed the terms of service user status updated information if successful, error otherwise
+ * @returns {Promise<Object>} A promise resolving to the updated terms of service user status
+ */
+TermsOfService.prototype.updateUserStatuses = function(termsOfServiceUserStatusesID, isAccepted, callback) {
+	var params = {
+		body: {
+			is_accepted: isAccepted
+		}
+	};
+
+	var apiPath = urlPath(USER_STATUSES_PATH, termsOfServiceUserStatusesID);
+	return this.client.wrapWithDefaultHandler(this.client.put)(apiPath, params, callback);
+};
+
+/**
+ * Creates a user status for terms of service, if already exists then update existing user status for terms of service
+ *
+ * API Endpoint: '/terms_of_service_user_statuses'
+ * Method: POST/PUT
+ *
+ * @param {string} termsOfServicesID - Terms of services ID to retrieve user statuses on
+ * @param {bool} isAccepted - Determines wehether the terms of services has been accepted or rejected
+ * @param {Object} [options] - Additional options. Can be left null in most cases.
+ * @param {string} [options.userID] - Optional, user id to retrieve terms of service status on, default is current user
+ * @param {Function} [callback] - Passed the terms of service user status information if successful, error otherwise
+ * @returns {Promise<Object>} A promise resolving to the terms of service user status
+ */
+TermsOfService.prototype.setUserStatuses = function(termsOfServicesID, isAccepted, options, callback) {
+	var params = {
+		body: {
+			tos: {
+				tos_id: termsOfServicesID,
+				type: 'terms_of_service'
+			},
+			is_accepted: isAccepted
+		}
+	};
+
+	if (options.userID) {
+		var user = {
+			user_id: options.userID,
+			type: 'user'
+		};
+		params.body.user = user;
```

Comment:
[nit] You don't need to create a `user` variable here, and could assign the object directly to `params.body.user` instead